tests/ui/consts/issue-52060.rs

```diff
@@ -1,7 +1,8 @@
 // Regression test for https://github.com/rust-lang/rust/issues/52060
 // The compiler shouldn't ICE in this case
+//@ check-pass
+
 static A: &'static [u32] = &[1];
```

Comment:
Please make this a `static mut`. Then it will keep erroring, which I think better preserves the original meaning of this test.

Reply:
Switched to `static mut`.

It now rightfully requires us to wrap the `A.len()` inside a `unsafe {..}` block. I am not sure if this will serve the original purpose of this test.

Reply:
Please add the `unsafe` block as well, then.

Reply:
The original test provoked this ICE output:
```
     Running `rustc --crate-name general_suffix src\lib.rs --crate-type lib --emit=dep-info,metadata -C debuginfo=2 -C metadata=167d6e50645cc7bb -C extra-filename=-167d6e50645cc7bb --out-dir W:\Rust\general-suffix\target\debug\deps -C incremental=W:\Rust\general-suffix\target\debug\incremental -L dependency=W:\Rust\general-suffix\target\debug\deps --extern ucd_raw=W:\Rust\general-suffix\target\debug\deps\libucd_raw-42a16cbe8e690df3.rmeta`
error[E0013]: constants cannot refer to statics, use a constant instead
 --> src\lib.rs:2:26
  |
2 | static B: [u32; 1] = [0; A.len()];
  |                          ^

error[E0394]: cannot refer to statics by value, use a constant instead
 --> src\lib.rs:2:26
  |
2 | static B: [u32; 1] = [0; A.len()];
  |                          ^ referring to another static by value
  |
  = note: use the address-of operator or a constant instead

error: `core::slice::<impl [T]>::len` is not yet stable as a const fn
 --> src\lib.rs:2:26
  |
2 | static B: [u32; 1] = [0; A.len()];
  |                          ^^^^^^^
  |
  = help: in Nightly builds, add `#![feature(const_slice_len)]` to the crate attributes to enable

thread 'main' panicked at 'already borrowed: BorrowMutError', libcore\result.rs:945:5
stack backtrace:
```

and then
```
query stack during panic:
#0 [const_eval] const-evaluating `A`
 --> src\lib.rs:2:26
  |
2 | static B: [u32; 1] = [0; A.len()];
  |                          ^
#1 [const_eval] const-evaluating `B::{{constant}}`
#2 [typeck_tables_of] processing `B`
end of query stack
error: aborting due to 3 previous errors
```

So it seems we're looking for something that indeed can't be used during const eval and thus causes an error.

Reply:
Thanks for the explanation! The test is now added to push the const eval further and reveal the actual eval failure.

Reply:
Thanks.

Can we rename this test to something like `illegal-static-refs-in-const-shouldnt-ice-issue-52060.rs`?

Reply:
I'm not particular to that name, so if you like anything else, by all means, just to make it clear what the test is testing for so we don't have to dig as deep in the future.

Reply:
I can go with something shorter, like `no-ice-from-static-in-const-issue-52060`? Then I will explain further in a comment at the beginning of the test about what happened with the ICE and the expectation, which is to print diagnostic instead of panicking.

Reply:
absolutely, sounds good to me.

Reply:
Renamed